src/CentOps.Api/Controllers/PublicParticipantController.cs

```diff
@@ -45,5 +47,26 @@ public async Task<ActionResult<ParticipantResponseModel>> Get(string id)
                 ? Ok(_mapper.Map<ParticipantResponseModel>(participant))
                 : NotFound(id);
         }
+
+        [HttpPut("state/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<ParticipantResponseModel>> Put(string id, [FromBody] ParticipantState newState)
+        {
+            try
+            {
+                var response = await _store.UpdateState(id, newState).ConfigureAwait(false);
+                return Ok(_mapper.Map<ParticipantStateReponseModel>(response));
```

Comment:
super nit: Just from a readability perspective, I always favour having a var for the thing you are returning separate from the return statement itself. I find it easier to read that way, could just be me though, feel free to ignore.
```suggestion
                var result = _mapper.Map<ParticipantStateReponseModel>(response);
                return Ok(result);
```

Reply:
Agreed just a short cut this, but readability would be a reason to commit this change